lib/sinon/sandbox.js

```diff
@@ -196,18 +196,15 @@ function Sandbox() {
     };
 
     sandbox.restoreContext = function restoreContext() {
-        var injectedKeys = sandbox.injectedKeys;
-        var injectInto = sandbox.injectInto;
-
-        if (!injectedKeys) {
+        if (!sandbox.injectedKeys) {
             return;
         }
 
-        forEach(injectedKeys, function (injectedKey) {
-            delete injectInto[injectedKey];
+        forEach(sandbox.injectedKeys, function (injectedKey) {
+            delete sandbox.injectInto[injectedKey];
         });
 
-        injectedKeys = [];
+        sandbox.injectedKeys = [];
```

Comment:
Do we actually need a new array?

```suggestion
        sandbox.injectedKeys.length = 0;
```